src/pyseekdb/client/collection.py

```
@@ -115,6 +115,30 @@ def sparse_embedding_function(self) -> Optional["SparseEmbeddingFunction"]:
+    def __dir__(self) -> list[str]:
+        attrs = super().__dir__()
+        if "refresh_index" in attrs and not self._is_refresh_available():
+            attrs.remove("refresh_index")
+        return attrs
+
+    def _is_refresh_available(self) -> bool:
+        checker = getattr(self._client, "_refresh_enabled", None)
+        if callable(checker):
+            try:
+                return bool(checker())
+            except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)